src/test/java/store/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,325 @@
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/categories/associate-categories-to-books.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/users/create-user.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/roles/create-roles.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/roles/add-roles-for-users.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/shoppingcarts/create-shopping-cart.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/cartitems/add-1984-to-cart-items.sql"));
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/cartitems/delete-1984-from-cart-items.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/shoppingcarts/delete-shopping-cart.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/roles/remove-roles-from-users.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/roles/delete-roles.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/users/delete-user.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/categories/remove-categories-from-books.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/categories/remove-categories.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/remove-books.sql"));
+        }
+    }
+
+    @Test
+    @DisplayName("Get shopping cart for user")
+    void getShoppingCartForUser_AuthenticatedUser_Success() throws Exception {
+        // Given
+        CartItemResponseDto item1 = new CartItemResponseDto()
+                .setId(1L)
+                .setBookId(1L)
+                .setBookTitle("1984")
+                .setQuantity(2);
+
+        ShoppingCartDto expected = new ShoppingCartDto()
+                .setId(1L)
+                .setUserId(1L)
+                .setCartItems(new HashSet<>(List.of(item1)));
+
+        String token = jwtUtil.generateToken("user@mail.com");
```

Comment:
you can use 
`@WithMockUser(username = "admin", roles = {"ADMIN"})`
above method